File: src/learning_rosbag/scripts/write_rosbag.py
#! /usr/bin/env python3

import rospy
import rosbag
from std_msgs.msg import String

if __name__ == "__main__":

    rospy.init_node("w_bag_p")

    bag = rosbag.Bag("/home/zwh/experiment_4/src/learning_rosbag/data/test.bag", 'w')

    s = String()
    s.data= "hahahaha"

    bag.write("chatter",s)
    bag.write("chatter",s)
    bag.write("chatter",s)

    bag.close()

File: src/learning_rosbag/src/write_rosbag.cpp
#include "ros/ros.h"
#include "rosbag/bag.h"
#include "std_msgs/String.h"

int main(int argc, char *argv[])
{
    ros::init(argc,argv,"bag_write");
    ros::NodeHandle nh;

    rosbag::Bag bag;

    bag.open("/home/zwh/experiment_4/src/learning_rosbag/data/test.bag", rosbag::BagMode::Write);

    std_msgs::String msg;
    msg.data = "hello world";
    bag.write("/chatter",ros::Time::now(),msg);
    bag.write("/chatter",ros::Time::now(),msg);
    bag.write("/chatter",ros::Time::now(),msg);
    bag.write("/chatter",ros::Time::now(),msg);

    bag.close();

    return 0;
}